artifacts/reader.py

```diff
@@ -227,6 +196,49 @@ def _ReadSupportedOS(self, definition_values, definition_object, name):
 
     definition_object.supported_os = supported_os
 
+  def _ReadSources(self, artifact_definition_values, artifact_definition):
+    """Reads the artifact definition sources
+
+    Args:
+      artifact_definition_values: the artifact definition as a dict.
+      artifact_definition_values: the artifact definition object (instance of
+                           ArtifactDefinition).
+
+    Raises:
+      FormatError: if the type indicator is not set or unsupported,
+                   or if required attributes are missing.
+    """
+    name = artifact_definition_values.get(u'name', None)
+    sources = artifact_definition_values.get(u'sources')
+    if not sources:
+      raise errors.FormatError(
+          u'Invalid artifact definition: {0} missing sources.'.format(name))
+
+    for source in sources:
+      type_indicator = source.get(u'type', None)
+      if not type_indicator:
+        raise errors.FormatError(
+            u'Invalid artifact definition: {0} source type.'.format(name))
+
+      attributes = source.get(u'attributes', None)
+
+      try:
+        source_type = artifact_definition.AppendSource(type_indicator,
+                                                       attributes)
+      except errors.FormatError as exception:
+        raise errors.FormatError(
+            u'Invalid artifact definition: {0}. {1}'.format(name, exception))
+
+      # TODO: deprecate these left overs from the collector definition.
```

Comment:
I think the _ReadSupportedOS would need to stay for the Sources, but I'm not sure if the others are intended to be valid anymore.